server/pulp/server/content/web/views.py

```diff
@@ -144,4 +156,13 @@ def get(self, request):
             return self.redirect(request, self.key)
 
         # NotFound
-        return HttpResponseNotFound(request.path_info)
+        raise Http404
+
+    @staticmethod
+    def directory_index(path):
+        listing = os.listdir(path)
+        context = {
+            'dirs': [f for f in listing if os.path.isdir(os.path.join(path, f))],
+            'files': [f for f in listing if os.path.isfile(os.path.join(path, f))]
```

Comment:
iterating over a list and doing the same work twice makes me sad, but I can be mollified by mixing in a builtin...
```python
listing = [os.path.join(path, f) for f in listing]
# ...
'dirs': filter(os.path.isdir, listing),
'files': filter(os.path.isfile, listing),
# snooty trailing comma on  final  item ^ :D
```